src/app/entry/current-collections/current-collections.component.ts

```diff
@@ -17,7 +17,7 @@ import { OrgLogoService } from '../../shared/org-logo.service';
 export class CurrentCollectionsComponent implements OnInit, OnChanges {
   @Input() id: number;
   // This is bad, should not have an input that's only passed down the chain
-  @Input() versions: Version[];
+  @Input() versions: WorkflowVersion[] | Tag[];
```

Comment:
It's a good idea, but I'm not sure if it's (easily) possible with the code generators we are using. The `@ApiModel` is a Swagger annotation, which we don't use any more; but the `@Schema` annotation looks like it has some attributes that could apply, `subType`, `oneOf`, ... I tried briefly with `oneOf`, and it generated a different openapi.yaml, but didn't seem to change the TS generated code.

@denis-yuen any thoughts or memories about this? I vaguely recall some discussion of hierarchy with Swagger-generated classes, perhaps with Gary, but the memories are so vague, I'm not sure they're real.

Reply:
Hmmm, I'm not entirely sure how this works myself across languages. 
But taking a look at `Workflow.java` for example. `@ApiModel` is definitely swagger, `@JsonTypeInfo` and `@JsonSubTypes` are jackson. However, The latter two do survive the code generation process and make it into the code generated `Workflow.java`. However, that just lets Jackson know which child class you're de-serializing, it doesn't let you address the object as `Entry` (I think)

I do think it could be worth tinkering with the `oneOf` or similar https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/ (open api doc) 

Is it possible that what you did with `oneOf` was correct, but just that the TS generated code was not up-to-par? Maybe we can look at the Java generated code from the same openapi.yaml

Reply:
dockstore/dockstore#5605